Modules/CalcPerform.lua

```diff
@@ -1595,7 +1886,603 @@ function calcs.perform(env)
+			env.player.mainSkill.infoTrigger = triggerName
+			env.player.mainSkill.skillFlags.dontDisplay = true
+		end
+	end
+
+	-- Helmet Focus Trigger
+	if env.player.mainSkill.skillData.triggeredByFocus and not env.player.mainSkill.skillFlags.minion then
+		local triggerName = "Focus"
+		local spellCount = 0
+		local icdr = calcLib.mod(env.player.mainSkill.skillModList, env.player.mainSkill.skillCfg, "FocusCooldownRecovery")
+		local trigRate = 0
+		local source = env.player.modDB:Flag(nil, "Condition:Focused")
+		for _, skill in ipairs(env.player.activeSkillList) do
+			if skill.skillData.triggeredByFocus and env.player.mainSkill.socketGroup.slot == skill.socketGroup.slot then
+				spellCount = spellCount + 1
+			end
+		end
+		if not source or spellCount < 1 then
+			env.player.mainSkill.skillData.triggeredByFocus = nil
+			env.player.mainSkill.infoMessage = s_format("No %s Triggering Skill Found", triggerName)
+			env.player.mainSkill.infoMessage2 = "DPS reported assuming Self-Cast"
+			env.player.mainSkill.infoTrigger = ""
+		else
+			env.player.mainSkill.skillData.triggered = true
+
+			output.ActionTriggerRate = getTriggerActionTriggerRate(env.player.mainSkill.skillData.cooldown, env, breakdown, true)
+
+			-- Get action trigger rate
+			local skillFocus = env.data.skills["Focus"]
+			local focusCD = skillFocus.levels[1].cooldown
+
+			trigRate = icdr / focusCD
+			output.SourceTriggerRate = trigRate
+			output.ServerTriggerRate = m_min(output.SourceTriggerRate, output.ActionTriggerRate)
+			if breakdown then
+				local modActionCooldown = focusCD / icdr
+				local rateCapAdjusted = m_ceil(modActionCooldown * data.misc.ServerTickRate) / data.misc.ServerTickRate
+				breakdown.SimData = {
+					s_format("%.2f ^8(base cooldown of focus trigger)", focusCD),
+					s_format("/ %.2f ^8(increased/reduced cooldown recovery)", icdr),
+					s_format("= %.4f ^8(final cooldown of trigger)", modActionCooldown),
+					s_format(""),
+					s_format("%.3f ^8(adjusted for server tick rate)", rateCapAdjusted),
+					s_format(""),
+					s_format("Trigger rate:"),
+					s_format("1 / %.3f", rateCapAdjusted),
+					s_format("= %.2f ^8per second", 1 / rateCapAdjusted),
+				}
+				breakdown.ServerTriggerRate = {
+					s_format("%.2f ^8(smaller of 'cap' and 'skill' trigger rates)", output.ServerTriggerRate),
+				}
+			end
+
+			-- Account for Trigger-related INC/MORE modifiers
+			addTriggerIncMoreMods(env.player.mainSkill, env.player.mainSkill)
+			env.player.mainSkill.skillData.triggerRate = output.ServerTriggerRate
+			env.player.mainSkill.skillData.triggerSource = source
+			env.player.mainSkill.infoMessage = "Focus Triggering Skill Found"
+			env.player.mainSkill.infoTrigger = triggerName
+			env.player.mainSkill.skillFlags.dontDisplay = true
+		end
+	end
+
+	-- Unique Item Trigger
+	if env.player.mainSkill.skillData.triggeredByUnique and not env.player.mainSkill.skillFlags.minion then
```

Comment:
Be careful with this, because I think there are more `triggeredByUnique` mods out there than this handles.  It's okay if we don't handle them all right away, but I did notice Soulwrest, for example, has a trigger rate of 0.

Reply:
Agreed, and I'm happy to handle and add support for such when they get reported. Part of the reason I'm hoping we get PoB 2.0 out soon so I have time to fix up stuff prior to next league launch.